qutebrowser/javascript/focustools.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2016-2017 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+ *
+ * This file is part of qutebrowser.
+ *
+ * qutebrowser is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * qutebrowser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Functions for interacting with focus from the js side.
+ */
+
+"use strict";
+
+window._qutebrowser.focustools = (function() {
+    const funcs = {};
+
```

Comment:
I tried looking for a python signal for 'documentReady' to migrate this into python but I couldn't seem to find one. Trying to use the loadStarted signal results in the js being run in the previous page (rather than the current page). I could hook loadProgress, but that's hacky, so I'd rather not do that. Do you have any suggestions on what I could use as a hook to get this in python?